docs/docs/adrs/adr-006-denom-dos-fixes

```diff
@@ -0,0 +1,53 @@
+---
+sidebar_position: 2
+title: ADR Template
+---
+# ADR 006: Denom DOS fixes
+
+## Changelog
+* 5/9/2023: ADR created
+
+## Status
+
+Accepted
+
+## Context
+
+The provider and consumer modules are vulnerable to similar issues involving an attacker sending millions of denoms to certain addresses and causing the chain to halt. This ADR outlines both fixes since they are similar. Both fixes involve processing only denoms that are on a whitelist to avoid iterating over millions of junk denoms but have different requirements and are implemented in different ways.
+
+## Decision
+
+### Provider
+
+- Put the distribution module's FeePoolAddress back on the blocklist so that it cannot receive funds from users.
+- Create a new address called ConsumerRewardPool and unblock it, allowing funds to be sent to it.
+- Create a set of strings in the database for ConsumerRewardDenoms.
+- Create an endpoint called RegisterConsumerRewardDenom which deducts a fee from the sender's account, sends it to the community pool and adds a string to the ConsumerRewardDenoms set.
+- Create a parameter called ConsumerRewardDenomRegistrationFee which determines the fee that is 
+- Create a function called TransferRewardsToFeeCollector which gets the entire ConsumerRewardDenoms set from the database, iterates over it, and for each entry:
+  - Gets the balance of this denom for the ConsumerRewardPool account
```

Comment:
Here you call the "strings" denoms. Maybe you should call them at all time denoms, i.e.,
> Create a set of denoms in the database for ConsumerRewardDenoms.

Reply:
Well, they are strings